libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1633,8 +1633,9 @@ string YulUtilFunctions::copyByteArrayToStorageFunction(ArrayType const& _fromTy
 
 				let oldLen := <byteArrayLength>(sload(slot))
 
+				let srcOffset := 0
```

Comment:
This would not have been necessary, but I'm introducing it as preparation for memory objects.